### PR TITLE
Add a setting to courses to limit registration to institutional accounts

### PR DESCRIPTION
**Pull request by jorg-vr**
This pull request adds a setting to courses to limit registration to institutional accounts. (Thus excluding personal accounts). This is also the new default setting for courses.

image

What a personal account sees:
image

- [x] Tests were added

Part of #2597
Closes #3802

### CODE REVIEW

File: config/locales/views/courses/en.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ en:
       registration-open_for_all-help_html: Anyone can register for this course.
       registration-open_for_institution-help_html: Only members of <span class="fill-institution">the configured institution</span> can register for this course.
       registration-closed-help_html: All new registrations are disabled.
+      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a private account, such as a personal gmail or outlook account, cannot register.
```
Comment (niknetniko) [inline config/locales/views/courses/en.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a private account, such as a personal Gmail or Outlook account, cannot register.
```

File: config/locales/views/courses/en.yml [hunk 2]
```diff
@@ -98,6 +100,7 @@ en:
       registration-open_for_all-info: "Registrations are open for everyone: all users can register for this course."
       registration-open_for_institution-info: "Registrations are open for members of %{institution}: only they can register for this course."
       registration-closed-info: "Registrations are closed: users can't register until you change the registration settings. Users who are currently registered still have access."
+      registration-open_for_institutional_users-info: "Registrations are open for users that belong to an institution: only they can register for this course. Registrations are closed for users who signed in using a private account, such as a personal gmail or outlook account."
```
Comment (niknetniko) [inline config/locales/views/courses/en.yml:103]:
```suggestion
      registration-open_for_institutional_users-info: "Registrations are open for users that belong to an institution: only they can register for this course. Registrations are closed for users who signed in using a private account, such as a personal Gmail or Outlook account."
```

File: config/locales/views/courses/nl.yml [hunk 3]
```diff
@@ -17,6 +17,7 @@ nl:
       registration-open_for_all-help_html: Iedereen kan zich voor deze cursus registreren.
       registration-open_for_institution-help_html: Enkel gebruikers van <span class="fill-institution">de ingestelde onderwijsinstelling</span> kunnen zich voor deze cursus registreren.
       registration-closed-help_html: Er zijn geen nieuwe registraties mogelijk.
+      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een privaat account, zoals een persoonlijk gmail of outlook account, kunnen niet registreren.
```
Comment (niknetniko) [inline config/locales/views/courses/nl.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account, kunnen niet registreren.
```

File: config/locales/views/courses/nl.yml [hunk 4]
```diff
@@ -107,6 +109,7 @@ nl:
       registration-open_for_all-info: 'De registraties staan open voor iedereen: gebruikers die op de registratie-knop klikken worden geregistreerd.'
       registration-open_for_institution-info: 'De registraties staan open voor gebruikers van %{institution}: enkel zij kunnen zich registreren voor deze cursus.'
       registration-closed-info: 'De registraties zijn gesloten: niemand kan zich nog registreren tot de registratie-instelling wordt aangepast. Gebruikers die geregistreerd zijn behouden hun toegang.'
+      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren: enkel zij kunnen zich registreren voor deze cursus. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privaat account, zoals een persoonlijk gmail of outlook account.'
```
Comment (niknetniko) [inline config/locales/views/courses/nl.yml:112]:
```suggestion
      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren: enkel zij kunnen zich registreren voor deze cursus. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account.'
```

**Review by niknetniko — CHANGES_REQUESTED** [review]
Looks good to me (I assume #3802 is nog longer needed then?).

Do we also want to migrate existing courses to "institutional only"?

**Comment by jorg-vr** [comment]
@niknetniko This is indeed a replacement of #3802 so I will close that pr once this one is approved

With the current code all existing courses which where `open for all` will be migrated to `open for institutional` because I changed the meaning of the enum value 0

**Review by niknetniko — APPROVED** [review]

File: config/locales/views/courses/en.yml [hunk 5]
```diff
@@ -98,6 +100,7 @@ en:
       registration-open_for_all-info: "Registrations are open for everyone: all users can register for this course."
       registration-open_for_institution-info: "Registrations are open for members of %{institution}: only they can register for this course."
       registration-closed-info: "Registrations are closed: users can't register until you change the registration settings. Users who are currently registered still have access."
+      registration-open_for_institutional_users-info: "Registrations are open for users that belong to an institution: only they can register for this course. Registrations are closed for users who signed in using a private account, such as a personal Gmail or Outlook account."
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:103]:
```suggestion
      registration-open_for_institutional_users-info: "Registrations are open for users that belong to any institution. Registrations are closed for users who signed in using a private account, such as a personal Gmail or Outlook account."
```

File: config/locales/views/courses/nl.yml [hunk 6]
```diff
@@ -17,6 +17,7 @@ nl:
       registration-open_for_all-help_html: Iedereen kan zich voor deze cursus registreren.
       registration-open_for_institution-help_html: Enkel gebruikers van <span class="fill-institution">de ingestelde onderwijsinstelling</span> kunnen zich voor deze cursus registreren.
       registration-closed-help_html: Er zijn geen nieuwe registraties mogelijk.
+      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account, kunnen niet registreren.
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Iedereen die tot een instelling behoort kan zich voor deze cursus registreren. Gebruikers die zijn ingelogd met een <b>privéaccount</b>, zoals een persoonlijk Gmail- of Outlook-account, kunnen zich <b>niet</b> registreren.
```

File: config/locales/views/courses/nl.yml [hunk 7]
```diff
@@ -107,6 +109,7 @@ nl:
       registration-open_for_all-info: 'De registraties staan open voor iedereen: gebruikers die op de registratie-knop klikken worden geregistreerd.'
       registration-open_for_institution-info: 'De registraties staan open voor gebruikers van %{institution}: enkel zij kunnen zich registreren voor deze cursus.'
       registration-closed-info: 'De registraties zijn gesloten: niemand kan zich nog registreren tot de registratie-instelling wordt aangepast. Gebruikers die geregistreerd zijn behouden hun toegang.'
+      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren: enkel zij kunnen zich registreren voor deze cursus. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account.'
```
Comment (bmesuere) [inline config/locales/views/courses/nl.yml:112]:
```suggestion
      registration-open_for_institutional_users-info: 'De registraties staan open voor gebruikers die tot een instelling behoren. De registraties zijn gesloten voor gebruikers die zijn ingelogd met een privéaccount, zoals een persoonlijk Gmail- of Outlook-account.'
```

File: config/locales/views/courses/en.yml [hunk 8]
```diff
@@ -17,6 +17,7 @@ en:
       registration-open_for_all-help_html: Anyone can register for this course.
       registration-open_for_institution-help_html: Only members of <span class="fill-institution">the configured institution</span> can register for this course.
       registration-closed-help_html: All new registrations are disabled.
+      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a private account, such as a personal Gmail or Outlook account, cannot register.
```
Comment (bmesuere) [inline config/locales/views/courses/en.yml:20]:
```suggestion
      registration-open_for_institutional_users-help_html: Anyone that belongs to an institution can register for this course. Users that signed in using a <b>private account</b>, such as a personal Gmail or Outlook account, <b>cannot</b> register.
```

**Review by bmesuere — APPROVED** [review]
I've added some text suggestions. Shouldn't there also be a migration here to change the access level in the database?

**Comment by jorg-vr** [comment]
@bmesuere as stated above

> With the current code all existing courses which where open for all will be migrated to open for institutional because I changed the meaning of the enum value 0

**Pull request merged**